glotaran/model/model.py

```diff
@@ -34,12 +34,15 @@
 }
 
 default_dataset_properties = {
+    "clp_guidance": {"type": str, "allow_none": True},
```

Comment:
A little nitpicky: I personally like to have an explicit boolean switch like `is_clp_guide` instead of implicit enabling by setting this to not `None`. there might be more options for guidance in future and then implicit eanabling becomes a problem. so I think there should be a more explicit way to enable it.

Another idea, which need discussion: we could intruce a `type` property to datasetmodels, which defaults to `dataset`, but could be clp guide or e.g. a measured irf, But is for another PR, since that ois the whole problem of how to add extra data to your analysis.

Reply:
IMHO adding `is_clp_guide` would be superfluous and make the API more boilerplate cluttered, which this PR tries to reduce. Rather than cluttering the API, I would inform the user that they did something wrong if they do.

@jsnel what are your thoughts on this?

Reply:
The addition of an explicit boolean switch like is_clp_guide doesn't make sense to me, but I like the idea of introducing a type to dataset models at some point. However, I also agree this is best introduced in a subsequent PR where that type is actually needed. 

For now I think the current form of introduction is fine as-is. If needed it can later be easily deprecated in favor of the type argument. So instead of:
```yml
clp_guidance: s1
```
you would get
```yml
type: clp_guidance
compartments: s1
```

Reply:
To make it more generic we could even use
```yaml
type: clp_guidance
clp_label: s1
```